src/llamafactory/chat/hf_engine.py

```diff
@@ -86,22 +92,37 @@ def _process_args(
             and image is not None
             and not hasattr(processor, "image_seq_length")
             and template.image_token not in messages[0]["content"]
-        ):  # llava-like models
+            and model_args.visual_inputs_type in ["vision_tower","phi3v_like"]
+        ):
+            # llava-like models
             messages[0]["content"] = template.image_token + messages[0]["content"]
+        elif image is not None and model_args.visual_inputs_type == "qwenvl_like":
```

Comment:
If only qwen-vl uses this image processor, why not just use the template name to represent
glm4v has the same problem